examples/window_ondemand.rs

```diff
@@ -0,0 +1,95 @@
+#![allow(clippy::single_match)]
+
+// Limit this example to only compatible platforms.
+#[cfg(any(windows_platform, macos_platform, x11_platform, wayland_platform,))]
+fn main() -> Result<(), impl std::error::Error> {
+    use simple_logger::SimpleLogger;
+
+    use winit::{
+        event::{Event, WindowEvent},
+        event_loop::EventLoop,
+        platform::run_ondemand::EventLoopExtRunOnDemand,
+        window::{Window, WindowBuilder},
+    };
+
+    #[derive(Default)]
+    struct App {
+        window: Option<Window>,
+    }
+
+    SimpleLogger::new().init().unwrap();
+    let mut event_loop = EventLoop::new();
+
+    {
+        let mut app = App::default();
+
+        event_loop.run_ondemand(move |event, event_loop, control_flow| {
+            control_flow.set_wait();
+            println!("Run 1: {:?}", event);
+
+            if let Some(window) = &app.window {
+                match event {
+                    Event::WindowEvent {
+                        event: WindowEvent::CloseRequested,
+                        window_id,
+                    } => {
+                        if window_id == window.id() {
+                            control_flow.set_exit()
```

Comment:
In this case isn't it ok, since the `App` will be dropped at the end of the first scope/block for the first `run_ondemand` which will drop the Window?

More generally yeah it's documented that you shouldn't try to preserve Window state between runs of `run_ondemand`, but it's not currently enforced in any technical way.

It could be nice to add some technical checks but atm most backends don't actually track a list of Windows internally so we don't really have a handle on their lifetime.

I figured initially it could be ok to just assert this in the documentation (considering wanting to at least provide a better API than `run_return` for this use case)

I don't really know how fiddly that would be to enforce this automatically for all backends.

I guess for adding checks it could help if all backends tracked a map of `Arc<>` window references and then it would be possible to check that the ref-count for all windows is one before exiting.

Reply:
@rib but the user will have the window in their struct, you don't remove it from here? I'd suggest to set it to None so it's explicitly dropped by the user this way.

Reply:
Okey, yeah can aim to unset before exiting the loop.

I was thinking it was going to be annoying to change because I was guessing the `app.window` borrow was going to stop me setting `app.window = None` but apparently the borrow checker is fine with that, so it was simpler to change than expected.